roachpb/data.go

```diff
@@ -623,7 +623,7 @@ func (t *Transaction) Restart(userPriority, upgradePriority int32, timestamp Tim
 // others) for the transaction. If t.ID is empty, then the transaction is
 // copied from o.
 func (t *Transaction) Update(o *Transaction) {
-	if o == nil {
+	if o == nil || len(o.ID) == 0 {
```

Comment:
Why did this change? When would we see a non-nil Transaction without a valid ID?